File: src/dataset/python/difference-of-squares_exercism-ThomasZumsteg/openAI/ChatGPT4_difference_of_squares_v1.py
def square_of_sum(num):
    """Square of sum of numbers less than or equal to [num]"""
    return (num * (num + 1) // 2) ** 2

def sum_of_squares(num):
    """Sum of squares of numbers less than or equal to [num]"""
    return num * (num + 1) * (2 * num + 1) // 6

def difference(num):
    """Difference between the square_of_sum and sum_of_squares"""
    return (num * (num + 1) * (num - 1) * (3 * num + 2)) // 12

File: src/dataset/javascript/difference-of-squares_exercism-ThomasZumsteg/openAI/ChatGPT4_difference-of-squares_v1.js
class Squares {
  constructor(num) {
    const numPlus1 = num + 1;
    const numPlus1Squared = numPlus1 * numPlus1;
    this.squareOfSums = (num * numPlus1Squared) / 4;
    this.sumOfSquares = (num * numPlus1 * (2 * num + 1)) / 6;
    this.difference = this.squareOfSums - this.sumOfSquares;
  }
}

module.exports = Squares;